tao_compiler/mlir/disc/transforms/disc_passes.td

```diff
@@ -406,4 +406,15 @@ def DiscStripShapeConstraintOpsPass : Pass<"disc-strip-shape-constraint-ops", "M
 def ForLoopUnrollInterleave : Pass<"disc-for-loop-unroll-interleave", "mlir::func::FuncOp"> {
   let summary = "Unroll SCF for loops and interleave the unrolling";
   let constructor = "createForLoopUnrollInterleavePass()";
+}
+
+def DiscDuplicateComputationForFusionPass : Pass<"disc-duplicate-computation-for-fusion", "mlir::func::FuncOp"> {
+  let summary = "Duplicate some computations in order to enable more fusion opportunity and remove memory footprint.";
+  let constructor = "createDiscDuplicateComputationForFusionPass()";
+  let options = [
+    Option<"gpu_enabled_", "gpu-enabled", "bool",
+           /*default=*/"true", "whether gpu is available.">,
+    Option<"fusion_strategy_", "fusion-strategy", "std::string",
```

Comment:
Do we actually need the `fusion-strategy ` option? If it is always  `base`, we can remove it in this PR, and add this option if it's required.

Reply:
I prefer to leave the option here even though we do not use it now actually. It's better if we make use of such config. Currently implementation is just a conservative strategy.  Furthermore, we only duplicate scalar-bcast pattern in this PR, while eventually we will need a general "duplicate fusion" pass like XLA.